wp/lib/bap_wp/src/environment.ml

```diff
@@ -205,15 +206,16 @@ let init_loop_unfold (num_unroll : int) : loop_handler =
   }
 
 let mk_env
-    ?subs:(subs = Seq.empty)
-    ?exp_conds:(exp_conds = [])
-    ?freshen_vars:(freshen = false)
-    ?arch:(arch = `x86_64)
+    ~subs:(subs : Sub.t Seq.t)
     ~specs:(specs : (Sub.t -> Arch.t -> fun_spec option) list)
     ~default_spec:(default_spec : Sub.t -> Arch.t -> fun_spec)
     ~jmp_spec:(jmp_spec : jmp_spec)
     ~int_spec:(int_spec : int_spec)
+    ~exp_conds:(exp_conds : exp_cond list)
     ~num_loop_unroll:(num_loop_unroll : int)
+    ~arch:(arch : Arch.t)
+    ~freshen_vars:(freshen : bool)
+    ~fun_input_regs:(fun_input_regs : bool)
     (ctx : Z3.context)
```

Comment:
Any hope of having a simple "default environment" here, like for `mk_ctxt` in Z3?

Reply:
Ok I'm an idiot, I forgot we had exactly that in `precondition.ml`.

Reply:
Is there a reason to have the parameter named freshen_vars and the variable freshen?

Reply:
I think we should be consistent. I'll change it to `freshen_vars`